### PR TITLE
Fix no delete share memory

### PR DESCRIPTION
**Pull request by wudi**
Missing call `shaop_delele()` in `getExceptions`  when the closure function is done without any exceptions in PcntlAdapter.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -1,9 +1,9 @@
 {
-    "name": "duncan3dc/fork-helper",
+    "name": "eaglewu/fork-helper",
```
Comment (duncan3dc) [inline composer.json:2]: @eaglewu It looks like you've accidentally pushed some other changes here.

File: src/Fork.php [hunk 2]
```diff
@@ -80,13 +80,13 @@ public function wait($pid = null)
             unset($this->threads[$pid]);
         }
 
+        $exceptions = $this->adapter->getExceptions();
```
Comment (duncan3dc) [inline src/Fork.php:83]: @eaglewu It's a little odd to call this here, what do you think about handling it in [SharedMemory](https://github.com/duncan3dc/fork-helper/blob/a253b551dfd98ef70307a2858c6b39cab4431235/src/SharedMemory.php#L8) using a [destructor](http://php.net/manual/en/language.oop5.decon.php#language.oop5.decon.destructor)?

**Comment by duncan3dc** [comment]
@eaglewu Thanks for contributing, it seems like a good idea to clean up the memory after use, but I'm not sure this is the best way to go about it, let me take a look

**Review by duncan3dc — CHANGES_REQUESTED** [review]

File: src/Fork.php (hunk 2 above)
Comment (wudi) [reply src/Fork.php:83]: Sorry! my mistake, i will push new request shot now.

**Comment by duncan3dc** [comment]
@eaglewu You don't need to create a new pull request, you can use [git rebase](https://help.github.com/articles/about-git-rebase/) to change the commits you've already pushed and remove ones you no longer want

**Comment by wudi** [comment]
@duncan3dc I got some other problems with the commit, Is not the best way to go about it,So let me take a look.

**Comment by sergeyklay** [comment]
@eaglewu ping

**Comment by martijnhartlief** [comment]
After using this for a while I get this:
`Warning: shmop_open(): unable to attach or create shared memory segment 'No space left on device' `

When you call:
`$ ipcs -m`
You'll see a huge list of memory sections claimed by the forker.
Even restarting PHP-FPM won't free the memory.

Any chance this can be picked up again?

**Comment by duncan3dc** [comment]
Hi @martijnhartlief, please open a new issue to report your problem, then we can investigate it there

**Comment by duncan3dc** [comment]
This is fixed in version `2.2.0` now, as long as you call `wait()` the shared memory will be deleted

**Pull request closed without merging**